integrated_channels/moodle/migrations/0031_auto_20240116_1158.py

```diff
@@ -0,0 +1,98 @@
+# Generated by Django 3.2.23 on 2024-01-16 11:58
+
+from django.db import migrations, models
+import fernet_fields.fields
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('moodle', '0030_merge_0028_auto_20231116_1826_0029_auto_20231106_1233'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='historicalmoodleenterprisecustomerconfiguration',
+            name='decrypted_password',
+        ),
+        migrations.RemoveField(
+            model_name='historicalmoodleenterprisecustomerconfiguration',
+            name='decrypted_token',
+        ),
+        migrations.RemoveField(
+            model_name='historicalmoodleenterprisecustomerconfiguration',
+            name='decrypted_username',
+        ),
+        migrations.RemoveField(
+            model_name='moodleenterprisecustomerconfiguration',
+            name='decrypted_password',
+        ),
+        migrations.RemoveField(
+            model_name='moodleenterprisecustomerconfiguration',
+            name='decrypted_token',
+        ),
+        migrations.RemoveField(
+            model_name='moodleenterprisecustomerconfiguration',
+            name='decrypted_username',
```

Comment:
Have we tried to run this migration locally on some test data to ensure no data is lost during the process? I believe removing the field may cause data loss. We should look online if Django provides a solution for this. Apparently there is a model state and then there is a db state. We'll have to rename the field at both levels. https://b0uh.github.io/django-model-change-field-name-with-no-db-impact.html

Reply:
https://stackoverflow.com/questions/3498140/migrating-django-model-field-name-change-without-losing-data

Reply:
@sameenfatima78 yes the auto-generated migration was prone to data loss,
I have restructured the migrations to prevent data loss, the approach i am using seemed simplest to me
I have used two separate migrations so that first the unencrypted columns become "deprecated" and only after that the name of the the older unencrypted columns can get assigned encrypted columns